cogs/obliterate.py

```diff
@@ -254,6 +255,68 @@ async def on_error(self, interaction: discord.Interaction, error: Exception):
         print(error)
         traceback.print_tb(error.__traceback__)
 
+class AppreciationModal(discord.ui.Modal, title="Message of Appreciation"):
+    def __init__(self, bot, appreciate_players):
+        super().__init__()
+        self.bot = bot
+        self.appreciate_player = appreciate_players
+
+    message = discord.ui.TextInput(label='Message of appreciation', min_length=1, max_length=200, required=True, style=TextStyle.short)
+    anonymous = discord.ui.TextInput(label='Would you like this post to be anonymous?', min_length=1, max_length=3, required=True, placeholder="Yes/No", style=TextStyle.short)
+
+    async def on_submit(self, interaction: discord.Interaction):
+        user_id = int(interaction.message.embeds[0].footer.text.replace('User ID: ', ''))
+        member = await interaction.guild.fetch_member(user_id)
+        message = self.message.value
+        anonymous = self.anonymous.value
+        appreciate_players = self.appreciate_players
+
+        if not anonymous.upper() in ['NO', 'N', 'YES', 'Y']:
+            return f'Error invalid value for anonymous: `{anonymous}`'
+        else:
+            for i, opt in enumerate(['NO', 'N', 'YES', 'Y']):
+                if anonymous.upper() == opt and i < 2:
+                    anonymous = False
+                    break
+                elif anonymous.upper() == opt:
+                    anonymous = True
+                    break
+
+        if not member:
+            return 'Error: user not found'
+
+        # Update appreciation sheet on roster
+        agc = await self.bot.agcm.authorize()
+        ss = await agc.open_by_key(config['obliterate_roster_key'])
```

Comment:
Reminder before merging this: create a sheet and add the key to `data/config.json`.
Also: will these appreciations be tracked on the roster? Or should we use a separate sheet?

Reply:
I was thinking this could be tracked on a new sheet within the roster spreadsheet? Helps to keep things together without making the roster itself too busy

Reply:
Sure, that works!